spec/amazon_ses_mailer/message_spec.rb

```diff
@@ -1,34 +1,42 @@
 # frozen_string_literal: true
+require 'my_interceptor'
 
 RSpec.describe AmazonSesMailer::Message do
-  let(:template_name) { 'welcome_email' }
-  let(:options) { { template: template_name, reply_to: 'saba@gmail.com' } }
-  subject { described_class.new(template_name) }
-  describe '.ensure_array' do
-    context 'when string is given in option' do
-      it 'should return a string' do
-        AmazonSesMailer::Base.mail(options).deliver
-        expect(options[:reply_to]).to eq('saba@gmail.com')
+    let(:options) { { template: 'welcome_email', reply_to: 'user@example.com' } }
+    let(:delivery_proc) { {template: 'welcome_email'} }
+    subject { described_class.new(options,MyInterceptor,delivery_proc) }
+    
+    describe '.ses_client' do
+      it 'should call ses client API' do
+        expect(AmazonSesMailer::Message).to receive(:ses_client).and_return(:ses_client)
```

Comment:
This does not test that the returned object is an instance of `Aws::SESV2::Client`.